src/rgw/rgw_rest_swift.cc

```diff
@@ -597,21 +597,22 @@ static int get_swift_versioning_settings(
 {
   /* Removing the Swift's versions location has lower priority than setting
    * a new one. That's the reason why we're handling it first. */
-  const std::string vlocdel =
-    s->info.env->get("HTTP_X_REMOVE_VERSIONS_LOCATION", "");
-  if (vlocdel.size()) {
+  bool exists = s->info.env->exists("HTTP_X_REMOVE_VERSIONS_LOCATION");
+  if (exists) {
```

Comment:
Sorry! The openstack-swift can not disable object verioning by removing its X-Remove-Versions-Location metadata header by sending an empty key value.